util/session/sessionmanager.go

```diff
@@ -490,86 +494,155 @@ type TokenVerifier interface {
 	VerifyToken(token string) (jwt.Claims, string, error)
 }
 
-// WithAuthMiddleware is an HTTP middleware used to ensure incoming
-// requests are authenticated before invoking the target handler. If
-// disabled is true, it will just invoke the next handler in the chain.
+// WithAuthMiddleware is an HTTP middleware used to ensure incoming requests are authenticated before invoking the target handler.
+// If disabled is true, it will just invoke the next handler in the chain.
+// It checks for tokens in a configured header (for IAP JWTs) first, then falls back to cookies.
 func WithAuthMiddleware(disabled bool, authn TokenVerifier, next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if !disabled {
-			cookies := r.Cookies()
-			tokenString, err := httputil.JoinCookies(common.AuthCookieName, cookies)
-			if err != nil {
-				http.Error(w, "Auth cookie not found", http.StatusBadRequest)
-				return
+		if disabled {
+			next.ServeHTTP(w, r)
+			return
+		}
+
+		var tokenString string
+		var err error
+		ctx := r.Context()
+
+		// Attempt to get settings manager from the verifier if possible
+		// This assumes the TokenVerifier implementation might expose settings
+		var settingsMgr *settings.SettingsManager
+		if sm, ok := authn.(*SessionManager); ok {
+			settingsMgr = sm.settingsMgr
+		}
```

Comment:
Create another method on the interface, or provide another interface param to the middleware to `GetToken(r *http.Request) string`